### PR TITLE
Using gateway endpoint for collectibles

### PR DESCRIPTION
**Pull request by jpalvarezl**
Handles #1046 

Changes proposed in this pull request:
- Using endpoint from gateway for `collectibles`
- Adjusted mock calls verifications

@gnosis/mobile-devs

### CODE REVIEW

File: data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt [hunk 1]
```diff
@@ -39,9 +38,8 @@ class TokenRepository(
         })
     }
 
-    //FIXME: use client gateway (grouping and sorting will be done on the backend side)
     suspend fun loadCollectiblesOf(safe: Solidity.Address): List<Collectible> =
-        transactionServiceApi.loadCollectibles(safe.asEthereumAddressChecksumString())
+        gatewayApi.loadCollectibles(safe.asEthereumAddressChecksumString())
             .asSequence()
             .groupBy {
                 it.address
```
Comment (elgatovital) [inline data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt:45]: Does client gw perform grouping?

**Comment by elgatovital** [comment]
content of ressources/load_collectibles.json should be replaced with correct response data

**Comment by jpalvarezl** [comment]
> The client gateway relays the collectibles endpoint from the transaction service adding a layer of caching to the request and alleviating the transaction service.
> There are no changes to be made to the models returned by the endpoint. 

There is no change in the structure. The json is still valid.

**Comment by elgatovital** [comment]
I propose to put assets related models to models.assets package

**Comment by jpalvarezl** [comment]
Re-structuring the project is beyond the scope of the issue #1046 . If you would like to do that let's open a new issue for it.

**Comment by jpalvarezl** [comment]
I would also like to remind you that my time in this sprint is very limited :)

File: data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt (hunk 1 above)
Comment (jpalvarezl) [reply data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt:45]: Nope. Just caching. There is no change in the `DTO` classes. The response is the same as from the TX service, as specified in the ticket.

**Review by elgatovital — APPROVED** [review]

**Pull request merged**